reana_client/cli/test.py

```diff
@@ -0,0 +1,174 @@
+"""REANA client test commands."""
+
+import sys
+import click
+from reana_client.cli.utils import add_access_token_options, check_connection
+
+from reana_commons.gherkin_parser.parser import (
+    parse_and_run_tests,
+    AnalysisTestStatus,
+)
+from reana_commons.gherkin_parser.data_fetcher import DataFetcherInterface
+from reana_commons.gherkin_parser.errors import FeatureFileError
+from reana_client.printer import display_message
+from reana_client.api.client import (
+    list_files,
+    get_workflow_disk_usage,
+    get_workflow_logs,
+    get_workflow_status,
+    get_workflow_specification,
+    download_file,
+)
+from reana_client.utils import get_reana_yaml_file_path, load_validate_reana_spec
+
+
+class DataFetcherClient(DataFetcherInterface):
+    """Implementation of the DataFetcherInterface using reana_client.api.client methods."""
+
+    def list_files(
+        self, workflow, access_token, file_name=None, page=None, size=None, search=None
+    ):
+        """Return the list of files for a given workflow workspace."""
+        return list_files(workflow, access_token, file_name, page, size, search)
+
+    def get_workflow_disk_usage(self, workflow, parameters, access_token):
+        """Display disk usage workflow."""
+        return get_workflow_disk_usage(workflow, parameters, access_token)
+
+    def get_workflow_logs(
+        self, workflow, access_token, steps=None, page=None, size=None
+    ):
+        """Get logs from a workflow engine, use existing API function."""
+        return get_workflow_logs(workflow, access_token, steps, page, size)
+
+    def get_workflow_status(self, workflow, access_token):
+        """Get of a previously created workflow."""
+        return get_workflow_status(workflow, access_token)
+
+    def get_workflow_specification(self, workflow, access_token):
+        """Get specification of previously created workflow."""
+        return get_workflow_specification(workflow, access_token)
+
+    def download_file(self, workflow, file_path, access_token):
+        """Download the requested file if it exists."""
+        return download_file(workflow, file_path, access_token)
+
+
+@click.group(help="Workflow run test commands")
+def test_group():
+    """Workflow run test commands."""
+
+
+@test_group.command("test")
+@click.option(
+    "-w",
+    "--workflow",
+    default="workflow",
+    help="Name of workflow to be tested, default is 'workflow'",
+)
+@click.option(
+    "-n",
+    "--test_file",
+    default=None,
+    help="Gherkin file for testing properties of a workflow execution. Overrides files in reana.yaml if provided.",
+)
```

Comment:
Maybe we can accept multiple test files? But let's discuss it together.

Also regarding cosmetics, let's use a dash instead of an underscore in the option name (`--test-file`).